pkg/ryokaicommon/utils/toml/toml.go

```diff
@@ -0,0 +1,144 @@
+// Package toml provides utilities for manipulating TOML (Tom's Obvious, Minimal Language) configurations
+// within Go applications. It offers a straightforward API for updating variables within a TOML configuration
+// string, handling both base and tagged sections. The package is designed to facilitate the dynamic
+// modification of TOML configurations, allowing for the programmatic setting of configuration values
+// without the need for direct file manipulation or external TOML parsing libraries and
+// with the possibility of saving original comments
+//
+// Example Usage:
+// Given a TOML configuration string, the package can be
+// used to update the value of a configuration variable as follows:
+//
+//	tomlValue := toml.Value{
+//	    Tag:   "server",
+//	    Name:  "port",
+//	    Value: "8080",
+//	}
+//	updatedConfig, err := toml.SetTomlVar(tomlValue, configStr)
+//	if err != nil {
+//	    // Handle error
+//	}
+//
+// This utility is particularly useful for applications that need to modify configuration values on the fly,
+// based on dynamic conditions or user input, without relying on external configuration management tools.
+package toml
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+type (
+	VariableNotFoundError struct {
+		VariableName string
+		Tag          string
+	}
+
+	// Value represents a TOML configuration variable, including its tag (section),
+	// name, and value. The tag is optional for variables in the base section.
+	Value struct {
+		Tag   string
+		Name  string
+		Value string
+	}
+)
+
+func (e *VariableNotFoundError) Error() string {
+	return fmt.Sprintf("variable '%s' not found in tag '%s'", e.VariableName, e.Tag)
+}
+
+// SetTomlVar updates a specific variable within a TOML configuration string.
+// It takes a Value struct containing the tag, name, and new value of the variable
+// to be updated, along with the original TOML configuration string.
+// It returns the updated TOML configuration string or an error
+// if the variable could not be found or updated.
+func SetTomlVar(tomlValue Value, configStr string) (string, error) {
+	tag := strings.TrimSpace(tomlValue.Tag)
+	name := strings.TrimSpace(tomlValue.Name)
+	value := strings.TrimSpace(tomlValue.Value)
+
+	tag = formatTag(tag)
+
+	lines := strings.Split(configStr, "\n")
+
+	searchedLineIndex, err := searchNeededLine(tag, name, lines)
```

Comment:
`searchedLineIndex` -> `foundLineIndex`
as the function itself return int as rep of integer and err if fails. Logicaly it always find something.